sources/packages/api/src/database/entities/user.model.ts

```diff
@@ -1,35 +1,93 @@
-import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from "typeorm";
+import { ColumnNames, TableNames } from "../constant";
 import { BaseModel } from "./base.model";
+import { IdentityProviders } from "./identity-providers.type";
 
-@Entity({ name: "users" })
+@Entity({ name: TableNames.User })
 export class User extends BaseModel {
   @PrimaryGeneratedColumn()
   id: number;
-
+  /**
+   * Unique user name generated by Keyclock. Usually in
+   * a form of a guid and a suffix like @bceid, @idir, @bcsc.
+   */
   @Column({
     name: "user_name",
     nullable: false,
   })
   userName: string;
-
+  /**
+   * User email.
+   */
   @Column({
     name: "email",
     nullable: false,
   })
   email: string;
-
+  /**
+   * First name or given names of the user.
+   */
   @Column({
     name: "first_name",
   })
   firstName: string;
```

Comment:
firstname is nullable. So we can add nullable:true and firstname? here. I know it is not a part of the PR.

Reply:
I changed it. It caused some other changes to the other 2 methods, but it was just minor changes.